pkg/trait/builder.go

```diff
@@ -59,7 +63,11 @@ func (t *builderTrait) Configure(e *Environment) (bool, error) {
 }
 
 func (t *builderTrait) Apply(e *Environment) error {
-	builderTask := t.builderTask(e)
+	builderTask, err := t.builderTask(e)
+	if err != nil {
+		// Should we clean the failing integration kit as well?
```

Comment:
Ì have a doubt here. If the task fails (ie, a bad format property is provided), we are safely returning and showing a log on the operator side. However, the failing integration kit will hang around until a manual removal is performed. I wonder if in this situation we can mark the `IntegrationKit` in order to be cleaned. Is that possible? how to perform that? @nicolaferraro @astefanutti do you have any suggestion?

Reply:
By default, an event will be emitted:

https://github.com/apache/camel-k/blob/1b64912e2f1e67bb3dfe287d3ba65ddc5e50cf33/pkg/controller/integrationkit/integrationkit_controller.go#L252

In that case, the reconciliation request is requeued for retry. That is applicable to general errors, that may be recoverable, such as conflicting changes. In case of an error that we know cannot be recovered, the IntegrationKit phase must be set to `IntegrationKitPhaseError`.

Reply:
Thanks. I managed to fix the problem by setting an error status:
```
		e.IntegrationKit.Status.Phase = v1.IntegrationKitPhaseError
		e.IntegrationKit.Status.SetCondition("IntegrationKitPropertiesFormatValid", corev1.ConditionFalse,
			"IntegrationKitPropertiesFormatValid", fmt.Sprintf("One or more properties where not formatted as expected: %s", err.Error()))
                return nil
```
However, after 5 retries, I'd expect to see the condition with the error specified in the `IntegrationKit`, but I just see a generic error:
```
  status:
    conditions:
    - lastTransitionTime: "2021-05-20T09:07:43Z"
      lastUpdateTime: "2021-05-20T09:07:43Z"
      message: IntegrationPlatform (camel-k)
      reason: IntegrationPlatformAvailable
      status: "True"
      type: IntegrationPlatformAvailable
    digest: v-xWT3kPLcPJ2Lhu0eFac7eWDCCIZBCbZhIPQRWPRBxQ
    failure:
      reason: ""
      recovery:
        attempt: 5
        attemptMax: 5
        attemptTime: "2021-05-20T09:10:04Z"
      time: "2021-05-20T09:07:48Z"
    phase: Error
```
How can I provide that error message in the `IntegrationKit`. I think we must set the failure reason to let the user know why it failed.

Reply:
Nevermind, I've just realized I was forgetting to update the status. Fixed with last commit.